src/test/java/IntegrationTestIT.java

```diff
@@ -0,0 +1,236 @@
+
+public class IntegrationTestIT {
+    UUID sessionId;
+    UUID gameId;
+
+    @Before
+    public void setup() {
+        RestAssured.baseURI = "http://localhost:8080";
+    }
+
+    @BeforeEach
+    public void setSessionId() {
+        String returnedDataFromServer = given()
+                .contentType(ContentType.JSON)
+                .when()
+                .get("/new_game")
+                .then()
+                .statusCode(200)
+                .extract()
+                .asString();
+        try {
+            // Split the input string by "&"
+            String[] keyValuePairs = returnedDataFromServer.split("&");
+
+            // Initialize variables
+            UUID sessionIdFromServer = null;
+            UUID gameIdFromServer = null;
+
+            // Iterate over key-value pairs
+            for (String pair : keyValuePairs) {
+                // Split each pair by "="
+                String[] entry = pair.split("=");
+
+                // Check for "session_id" and "game_id" keys
+                if (entry.length == 2) {
+                    String key = entry[0];
+                    String value = URLDecoder.decode(entry[1], StandardCharsets.UTF_8.toString());
+
+                    if ("session_id".equals(key)) {
+                        sessionIdFromServer = UUID.fromString(value);
+                    } else if ("game_id".equals(key)) {
+                        gameIdFromServer = UUID.fromString(value);
+                    }
+                }
+            }
+            this.sessionId = sessionIdFromServer;
+            this.gameId = gameIdFromServer;
+
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void getRequestForNewGameWithoutSessionId() {
+        String returnedData = given()
+                .contentType(ContentType.JSON)
+                .when()
+                .get("/new_game")
+                .then()
+                .statusCode(200)
+                .extract()
+                .asString();
+        assertNotNull(returnedData);
+        assertTrue(!returnedData.isEmpty());
+        parsingKeys(returnedData);
+        assertNotEquals(IDs.get("session_id"), sessionId);
+        assertNotEquals(IDs.get("game_id"), gameId);
+    }
+
+    @Test
+    public void getRequestForNewGameWithSessionId() {
+        String returnedData = given()
+                .contentType(ContentType.JSON)
+                .queryParam("session_id", sessionId)
+                .when()
+                .get("/new_game")
+                .then()
+                .statusCode(200)
+                .extract()
+                .asString();
+        assertNotNull(returnedData);
+        assertTrue(!returnedData.isEmpty());
+        parsingKeys(returnedData);
+        assertEquals(IDs.get("session_id"), sessionId);
+        assertNotEquals(IDs.get("game_id"), gameId);
+    }
+
+    @Test
+    public void postMoveForSendWrongCoordinates() {
```

Comment:
Such tests look cool! Probably we will need to replace them since we are moving away from REST.